### PR TITLE
FEATURE: Logged out view

### PR DESCRIPTION
**Pull request by co012**
Screenshot 2022-09-21 at 12 18 35
Screenshot 2022-09-21 at 12 18 41
Screenshot 2022-09-21 at 12 19 08
Screenshot 2022-09-21 at 12 19 23

Needs https://github.com/skyhitz/backend/pull/54 to be merged

### CODE REVIEW

**Comment by co012** [comment]
@alejomendoza, I removed dashboard tabbar from desktop view in favour of navbar links. What do you think about it, should I put it back and hide navbar links or should I be showing both.

File: packages/app/ui/navigation/dashboardNavigation.tsx [hunk 1]
```diff
@@ -16,10 +15,7 @@ export function DashboardNavigation({
   return (
     <View className="flex flex-1 max-h-[100vh] bg-blue-dark overflow-hidden">
       <Navbar />
-      <View className="flex flex-row flex-1">
-        {tw.prefixMatch("sm") && <DashboardTabBar column />}
-        {children}
-      </View>
+      {children}
```
Comment (co012) [inline packages/app/ui/navigation/dashboardNavigation.tsx:18]: @alejomendoza, I removed dashboard tabbar from desktop view in favour of navbar links. What do you think about it, should I put it back and hide navbar links or should I be showing both.

Comment (alejomendoza) [reply packages/app/ui/navigation/dashboardNavigation.tsx:18]: Good question. Dashboard tabbar should be in place when logged in, navbar links only show up on logged out. 
Looking great awesome job.

**Comment by alejomendoza** [comment]
Screen Shot 2022-09-21 at 7 34 01 AM
Looking great, only one thing I found, there is a weird padding at the bottom of the nav when scrolling down.  I added a border so you could see it.

File: packages/app/utils/authGuard.tsx [hunk 2]
```diff
@@ -4,21 +4,39 @@ import { userAtom } from "app/state/user";
...
+type Props = {
+  children: ReactNode;
+  redirect?: boolean;
+  fallback?: ReactNode;
+};
+
+export function AuthGuard({
+  children,
+  redirect = true,
+  fallback = <SplashScreen />,
```
Comment (jmysliv) [inline packages/app/utils/authGuard.tsx:16]:
```suggestion
  fallback = SplashScreen,
```

File: packages/app/utils/authGuard.tsx [hunk 3]
```diff
@@ -4,21 +4,39 @@ import { userAtom } from "app/state/user";
 import { useRouter } from "solito/router";
 import { SplashScreen } from "app/features/splash/splashScreen";
 
-export function AuthGuard({ children }: { children: ReactNode }) {
+type Props = {
+  children: ReactNode;
+  redirect?: boolean;
+  fallback?: ReactNode;
```
Comment (jmysliv) [inline packages/app/utils/authGuard.tsx:10]:
```suggestion
  fallback?: () => ReactNode;
```

**Review by jmysliv — COMMENTED** [review]
I've noticed there are a lot of if conditions because of this feature. What do you think about creating a separate route for public users? I.ex `public/search`, `public/charts`? I think it would be more readable, but I'm open to discussion here.

File: packages/app/utils/authGuard.tsx (hunk 3 above)
Comment (co012) [reply packages/app/utils/authGuard.tsx:10]: If we do it like this we will lose the ability to use fallback with props

Comment (jmysliv) [reply packages/app/utils/authGuard.tsx:10]: I guess we don't need that option right now anyway.

File: packages/app/ui/navigation/dashboardTabBar.tsx [hunk 4]
```diff
@@ -25,17 +28,22 @@ export default function DashboardTabBar({ column }: { column?: boolean }) {
...
           source={require("app/assets/images/icon.png")}
         />
       </Link>
-      <Link viewProps={{ style: LinkStyle }} href={"/dashboard/profile"}>
-        <Icon name="account-outline" size={32} color="white" />
-      </Link>
+      {user && (
+        <Link
+          viewProps={{ style: LinkStyle }}
+          href={user ? "/dashboard/profile" : "/"}
```
Comment (jmysliv) [inline packages/app/ui/navigation/dashboardTabBar.tsx:42]:
```suggestion
          href={"/dashboard/profile"}
```

**Review by jmysliv — APPROVED** [review]
That one thing, and you can merge 🚀

**Pull request merged**